x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushStatsToSource.java

```diff
@@ -94,15 +94,19 @@ private Tuple<List<Attribute>, List<EsStatsQueryExec.Stat>> pushableStats(
                             // check if regular field
                             else {
                                 if (target instanceof FieldAttribute fa) {
-                                    var fName = fa.name();
+                                    var fName = fa.fieldName();
```

Comment:
Wow! Is this the fix? I imagine this could have impacts in many places, so could this fix other bugs we've not noticed?

Reply:
Yep, this is the fix. Simple oversight to use the correct field name - in the past, the field _attribute_ name and the field name were the same, but union types had to break with this pattern.

I do not see other situations that this may fix, too, because the specific stats-pushdown optimization only triggers in a narrow slice of queries, anyway.

Reply:
What I'm wondering is if we could make this mistake again and if we could prevent it.
Also, should we insert filters here using `#fieldName()`?
https://github.com/elastic/elasticsearch/blob/9e0a5af898b0d78c48fb22e758ae59fe5a1c8266/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/local/InferNonNullAggConstraint.java#L58